src/manifold/src/boolean3.cpp

```diff
@@ -578,6 +584,14 @@ Boolean3::Boolean3(const Manifold::Impl &inP, const Manifold::Impl &inQ,
       Intersect12(inQ, inP, s20, p2q0, s11, p1q1, z20, xyzz11, p2q1_, false);
   PRINT("x21 size = " << x21_.size());
 
+  if (p0q2.size() >= std::numeric_limits<int>::max() ||
+      p2q0.size() >= std::numeric_limits<int>::max() ||
```

Comment:
Why should 0-2 (vertices shadowing triangles) have to be fewer than INT_MAX? That's still worst case an O(n^2) combination problem. The other four below this are actually contributing verts, so the check makes more sense (though shouldn't it be their sum?).

Reply:
Indeed, I was thinking that `p0`/`q0` have the same size as `p0q2`/`p2q0`, and thought that they affect the size of `w03_` and `w30_`. Will address this.

Reply:
fixed, need to check if the current one is correct.